CMakeLists.txt

```diff
@@ -376,3 +377,32 @@ add_custom_target(
   COMMAND ${CMAKE_COMMAND} -DCOMPONENT=viam-cpp-sdk_examples -P ${CMAKE_BINARY_DIR}/cmake_install.cmake
   DEPENDS all
 )
+
+add_custom_target(
+  install-tests
+  COMMAND ${CMAKE_COMMAND} -DCOMPONENT=viam-cpp-sdk_tests -P ${CMAKE_BINARY_DIR}/cmake_install.cmake
+  DEPENDS all
+)
+
+enable_testing()
```

Comment:
By having `enable_testing` up here you are enabling it for the entire tree, not just the tests under the `test` directory, per the docs. I don't know what those effects are, but in general I'd recommend scoping effects as narrowly as possible. If you take my suggestion to move the `test` directory under `src`, then this `enable_testing` call would belong in `src/test/CMakeLists.txt`